src/main/java/org/openrefine/extensions/commons/functions/ExtractFromTemplate.java

```diff
@@ -0,0 +1,102 @@
+package org.openrefine.extensions.commons.functions;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.openrefine.extensions.commons.utils.WikitextParsingUtilities;
+import org.sweble.wikitext.parser.ParserConfig;
+import org.sweble.wikitext.parser.nodes.WtNode;
+import org.sweble.wikitext.parser.nodes.WtParsedWikitextPage;
+import org.sweble.wikitext.parser.nodes.WtTemplate;
+import org.sweble.wikitext.parser.nodes.WtTemplateArgument;
+import org.sweble.wikitext.parser.nodes.WtTemplateArguments;
+import org.sweble.wikitext.parser.utils.SimpleParserConfig;
+import org.sweble.wikitext.parser.utils.WtPrettyPrinter;
+
+import com.google.refine.expr.EvalError;
+import com.google.refine.expr.functions.Type;
+import com.google.refine.grel.ControlFunctionRegistry;
+import com.google.refine.grel.Function;
+
+import de.fau.cs.osr.ptk.common.AstVisitor;
+
+public class ExtractFromTemplate implements Function {
+
+    public class FindTemplateValues extends AstVisitor<WtNode> {
+
+        private String templateName;
+        private String paramName;
+        private List<String> values = new ArrayList<>();
+
+        public FindTemplateValues(String tName, String pName) {
+            this.templateName = tName;
+            this.paramName = pName;
+        }
+
+        public void visit(WtNode node) {
+            iterate(node);
+        }
+        public void visit(WtTemplate template) {
+            // only render templates if we are told to do so or inside a reference
+            if (templateName.equals(template.getName().toString())) {
+                WtTemplateArguments args = template.getArgs();
+                for (int i = 0; i != args.size(); i++) {
+                    WtTemplateArgument arg = (WtTemplateArgument) args.get(i);
+                    StringWriter writer = new StringWriter();
+                    WtPrettyPrinter treeToString = (WtPrettyPrinter) new WtPrettyPrinter(writer).go(arg);
```

Comment:
I just realized that there is an even simpler method to pretty-print a node: you can use the `WtPrettyPrinter.print(WtNode)` static method